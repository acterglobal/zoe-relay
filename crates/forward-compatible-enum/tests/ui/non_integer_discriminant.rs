use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
pub enum NonIntegerDiscriminant {
    #[discriminant("not_a_number")]  // Must be u32!
    Variant1,
    Unknown { discriminant: u32, data: Vec<u8> },
}