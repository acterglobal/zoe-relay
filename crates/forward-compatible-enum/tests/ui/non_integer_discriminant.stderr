error: expected integer literal
 --> tests/ui/non_integer_discriminant.rs:5:20
  |
5 |     #[discriminant("not_a_number")]  // Must be u32!
  |                    ^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/non_integer_discriminant.rs:8:2
  |
8 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/non_integer_discriminant.rs`
